src/Tizen.NUI.Components/Controls/AlertDialog.cs

```diff
@@ -17,154 +17,95 @@
 
 using System;
 using System.ComponentModel;
+using System.Collections.Generic;
 using Tizen.NUI.BaseComponents;
 
 namespace Tizen.NUI.Components
 {
-    /// <summary>
-    /// Types of the action button of AlertDialog.
-    /// </summary>
-    [EditorBrowsable(EditorBrowsableState.Never)]
-    public enum AlertDialogActionButtonType
-    {
-        /// <summary>
-        /// Type of the positive action button.
-        /// </summary>
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        Positive,
-
-        /// <summary>
-        /// Type of the negative action button.
-        /// </summary>
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        Negative
-    }
-
     /// <summary>
     /// AlertDialog class shows a dialog with title, message and action buttons.
     /// </summary>
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public class AlertDialog : Dialog
+    public class AlertDialog : Control
```

Comment:
I am not sure, but is the Dialog class removed?
I think Control is base class and AlertDialog seems to be a kind of Dialog class, so I think it looks better the previous change of AlertDialog inherits from Dialog. 
is there any reason to change the base class?

Reply:
Originally, Scrim (dimmed background area) was a property of Dialog. So AlertDialog was derived from Dialog.
But now, Scrim was a property of DialogPage and Dialog and AlertDialog only shows the popup content itself.

Dialog has only one content as follows.
- View Content

AlertDialog has three contents as follows.
- View TitleContent
- View Content
- View ActionContent

So to inherit Dialog, we have either 2 ways.
1. The positioning contents logic may be different between Dialog and AlertDialog
   Dialog positions Content at the center of the screen.
   AlertDialog positions 3 contents at the center of the screen.

OR

2. AlertDialog sets its 3 contents into Dialog.Content.
   (i.e. AlertDialog's TitleContent, BodyContent, ActionContent -> Dialog.Content)

I thought that AlertDialog may not be derived from AlertDialog because Scrim is no longer Dialog's property and it would be complicated the above.

Could you give me your opinion about the above?

Reply:
OK, I see. I think it's good to go. thank you for explanation.